### PR TITLE
Task runner and common dev tasks. Closes gh-118

### PR DESCRIPTION
**Pull request by rwaldron**
This is the first round. There will be a follow up issue to write some base line unit tests for the existing features. 

Additionally, automated ci runs with travis needs to be set up for this repo.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      options: {
+        jshintrc: '.jshintrc'
+      },
+      all: [
+        'bin/*',
+        'lib/**/*.js',
+        'test/**/*.js',
+        'Gruntfile.js',
+
+        // This is commented out because there are
```
Comment (johnnyman727) [inline Gruntfile.js]: @rwaldron would it help if I went through these errors?

File: test/unit/readme.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+TODO: 
+
+- write auto-mocks
+- install sinon
+- write units for existing functionality
```
Comment (johnnyman727) [inline test/unit/readme.md]: I can also help with this!

Comment (rwaldron) [inline test/unit/readme.md]: These "todo" items should be filed as new issues when this lands.

**Comment by johnnyman727** [comment]
@rwaldron I'm new to Grunt so let me know if I've configured something wrong, but when I run `grunt` in the root of the project, I get the following error: ```Warning: Task "grunt-jsbeautifier" not found. Use --force to continue.```. Is this expected?

File: lib/usb/usb_process.js [hunk 3]
```diff
@@ -13,47 +11,46 @@ function USBProcess(id, daemon) {
...
 
-  // This puts the stream into 'flowing mode' 
+  // This puts the stream into 'flowing mode'
   // so that the stream emits ('end') events without 'data' listeners
   this.stdout.resume();
   this.stderr.resume();
 
   this.stdout.signal = function(signalToSend) {
-    switch(signalToSend) {
+    // RWQ: Does this need to a `switch`?
```
Comment (johnnyman727) [inline lib/usb/usb_process.js]: It eventually will; I only have one case at the moment but I suspect there will be more signals that we'd like to send. In fact, #126 adds another case.

**Comment by johnnyman727** [comment]
Quick update: I started working through this PR but was having issues getting code deploys to work properly. Part of the issue is almost definitely that I'm using a new version of the USBDaemon firmware which handles USB comms but it may also be an issue with the bugs you fixed. Will dive in deeper tomorrow or Saturday.

**Comment by rwaldron** [comment]
Sorry for the delayed follow up, I had family things going on during the end of last week

**Comment by rwaldron** [comment]
There's been too much churn since this patch was created. I don't have the time resources right now to properly rebase and resolve all the conflicts. I'm going to take another approach, which is to do only one lint rule pass at a time.

**Comment by johnnyman727** [comment]
Yesterday I had the chance to dive into this PR on a plane ride and I found only two issues with the PR:

- [Line 84](https://github.com/tessel/t2-cli/pull/124/files#diff-f68567477d803b49930337bf7fe1556bL86) was incorrectly replaced with `return Promise.settle([usb.findConnections(), lan.findConnections(timeout)])` when it needs to be `return Promise.settle(connectionTypes)`.

- The `t2 init` command no longer works but it appears to be an issue with the Promzard library. I haven't investigated very far.

I'm happy to move my attention to the new PR but I'd thought I'd place a record of my review here.

**Pull request closed without merging**